vitality/__init__.py

```diff
@@ -84,12 +97,32 @@ def signup():
         if request.method == 'POST':
             session.pop('user_id', None)
             username = escape(request.form['username'])
+            if alphaPattern.search(username):
+                print("valid")
+            else:
+                raise Exception("Invalid characters")
             password = escape(request.form['password'])
             name = escape(request.form['name'])
+            if alphaPattern.search(password):
+                print("valid")
+            else:
+                raise Exception("Invalid characters")
             re_password = escape(request.form['repassword'])
             location = escape(request.form['location'])
+            if alphaPattern.search(re_password):
+                print("valid")
+            else:
+                raise Exception("Invalid characters")
             phone = escape(request.form['phone'])
+            if numberPattern.search(phone):
+                print("valid")
+            else:
+                raise Exception("Invalid characters")
             usertype = escape(request.form['usertype'])
+            if stringPattern.search(usertype):
+                print("valid")
+            else:
+                raise Exception("Invalid characters")
```

Comment:
Same as above: 
Create a new class which inherits Exception and raise that new class instead. 

Probably called InvalidCharactersError.